ktoml-core/src/commonTest/kotlin/com/akuleshov7/ktoml/decoders/multiline/BasicMultilineStringDecoderTest.kt

```diff
@@ -0,0 +1,236 @@
+package com.akuleshov7.ktoml.decoders.multiline
+
+import com.akuleshov7.ktoml.Toml
+import com.akuleshov7.ktoml.exceptions.MissingRequiredPropertyException
+import com.akuleshov7.ktoml.exceptions.ParseException
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+
+class BasicMultilineStringDecoderTest {
+
+    @Serializable
+    data class SimpleString(val a: String)
+
+    @Serializable
+    data class SimpleStrings(val a: String, val b: String)
+
+    @Serializable
+    data class StringAndInt(val a: String, val b: Int)
+
+    @Serializable
+    data class StringAndInts(val a: String, val b: Int, val c: Int)
+
+    @Test
+    fun testMultilineBasicStringDecode() {
+        var test = """
+            a = ""${'"'}abc""${'"'}
+        """.trimIndent()
+        assertEquals(SimpleString("abc"), Toml.decodeFromString(test))
+
+        test = """
+            a = ""${'"'}
+            first line
+            second line""${'"'}
+        """.trimIndent()
+        assertEquals(SimpleString("first line\nsecond line"), Toml.decodeFromString(test))
+
+        test = """
+            a = ""${'"'}first line
+            second line""${'"'}
+        """.trimIndent()
+        assertEquals(SimpleString("first line\nsecond line"), Toml.decodeFromString(test))
+
+        test = """
+            a = ""${'"'}first line
+
+            second line""${'"'}
+        """.trimIndent()
+        assertEquals(SimpleString("first line\n\nsecond line"), Toml.decodeFromString(test))
+
+        test = "a = \"\"\"first line\n\t  \nsecond line\"\"\""
+        assertEquals(SimpleString("first line\n\t  \nsecond line"), Toml.decodeFromString(test))
+
+        test = "a = \"\"\"\n" +
+            "Roses are red\n" +
+            "Violets are blue\"\"\""
+        assertEquals(SimpleString("Roses are red\nViolets are blue"), Toml.decodeFromString(test))
+
+        test = "a = \"\"\"" +
+            "\n\nTest\"\"\"" +
+            "\n\nb = 2"
+        assertEquals(
+            StringAndInt("\nTest", 2),
+            Toml.decodeFromString(test)
+        )
+
+        test = "a = \"\"\"" +
+            "\nTest" +
+            "\nb = 32\n\"\"\""
+        assertEquals(
+            SimpleString(a = "Test\nb = 32\n"),
+            Toml.decodeFromString(test)
+        )
+    }
+
+    @Test
+    fun lineEndingBackslash() {
+        var test = """
+            a = ""${'"'}
+            first line \
+            second line \
+            
+            third line""${'"'}
+        """.trimIndent()
+        assertEquals(SimpleString("first line second line third line"), Toml.decodeFromString(test))
+
+        test = "a = \"\"\"\n" +
+            "The quick brown \\\n\n\n" +
+            "  fox jumps over \\\n" +
+            "    the lazy dog.\"\"\""
```

Comment:
Why not use multiline strings here? If it's the quotes, when I have to write triple quotes I usually use templates similar to what you did above this. The strings here are bit hard to read

Something like this should work:

```kotlin
val tripleQuotes = "\"\"\""

test = """
a = $tripleQuotes
The quick brown \



  fox jumps over \
    the lazy dog.$tripleQuotes
""".trimIndex()
```

Reply:
Added